File: shared-code/rust/app/src/state.rs
use crate::{config::AppConfig, task::Task};
use std::collections::HashMap;
use uuid::Uuid;

pub struct AppState {
    pub tasks: HashMap<Uuid, Task>,
    pub config: AppConfig,
}

impl AppState {
    pub fn new(config: AppConfig) -> Self {
        Self {
            tasks: HashMap::new(),
            config,
        }
    }
}

File: shared-code/go/app/state.go
package app

import (
	"sync"

	"github.com/google/uuid"
)

// TODO: fields should be private
type AppState struct {
	Mu     sync.RWMutex
	Tasks  map[uuid.UUID]Task
	Config AppConfig
}

func NewState(appConfig AppConfig) *AppState {
	return &AppState{
		Tasks:  make(map[uuid.UUID]Task),
		Config: appConfig,
	}
}
